### DISCUSSION TITLE
gcc10 is not getting installed properly in mac

### DISCUSSION THREAD

**Discussion by Brijesh1in** [Homebrew/discussions#303] [Everyday usage] [answered]
Hello Community I recently installed gcc10 using homebrew and it’s not working . I am not able to compile anything .
Here is the terminal log .

![image](https://user-images.githubusercontent.com/33791137/102498426-f6d40580-409f-11eb-81cb-af59c0c9d2bf.png)
```cpp
#include <iostream>
using namespace std;

int main(){
	
	cout<<"Hell";
}
```
This was the code I had written . even if I am using `gcc something.c` then it’s not able to find stdio.h file .

**Comment by gromgit** [comment]
Try `brew reinstall -d -v gcc` and watch out for any errors, then try your compile again.

If that still doesn't work, post your `brew config` and `brew doctor` output.

**Reply by Brijesh1in** [reply]
![Screenshot 2020-12-17 at 8 57 51 PM](https://user-images.githubusercontent.com/33791137/102507672-88e10b80-40aa-11eb-8454-e3507768ad01.png)
It didn't work @gromgit :(

**Comment by carlocab** [comment]
Aside from the output requested by @gromgit above, please also show the output of `g++-10 -v test.cpp`. I suspect that `gcc` can't find the right headers because it's looking in the wrong directories.

**Comment by Brijesh1in** [comment]
![Screenshot 2020-12-17 at 9 00 02 PM](https://user-images.githubusercontent.com/33791137/102507960-d6f60f00-40aa-11eb-8d67-b6916cc6915b.png)
@carlocab Here I have uploaded the screenshot of the output .

**Reply by carlocab** [reply]
Thanks. But, next time someone asks you for the output of the command, please copy and paste the text. Screenshots are very difficult to read from.

**Reply by carlocab** [reply]
There's your problem, I think. It's looking for headers in `/Library/Developer/CommandLineTools/SDKs/MacOSX11.0.sdk/usr/include` when it needs to be looking in `/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/usr/include`.

I think `gcc` will need to be rebuilt. I've also reproduced this problem with `llvm` `clang++`.

If you have the patience for it, please try to do `brew reinstall -s gcc`. This will rebuild `gcc` from source, and will take quite some time. It would be a great help if you could do this, as it would help us verify that rebuilding `gcc` from source will in fact fix the problem.

**Reply by Brijesh1in** [reply]
Hi @carlocab sorry for the late response . I tried `brew reinstall -s gcc` but it is taking too much time . I think it has been around 10 hours and it got stuck on (AND I SLEPT)
```
==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/7baf6e2f/gcc/bigsur.diff
######################################################################## 100.0%
==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-10.2.0/gcc-10.2.0.tar.xz
######################################################################## 100.0%
==> Reinstalling gcc 
==> Patching
==> Applying bigsur.diff
patching file gcc/config/darwin-c.c
patching file gcc/config/darwin-driver.c
==> ../configure --build=x86_64-apple-darwin20 --prefix=/usr/local/Cellar/gcc/10.2.0 --libdir=/usr/local/Cellar/gcc/10.2.0/lib/gcc/10 --disable-nls --enable-checking=release --enable-languages=c,c++,objc,
==> make BOOT_LDFLAGS=-Wl,-headerpad_max_install_names
```
Sorry for the screenshot .

**Reply by carlocab** [reply]
Yes, that really will take a long time; sorry. I don't think it's stuck. It's just doing lots of things in the background that it's not displaying.

Please give it a few more hours (sorry).

**Reply by Brijesh1in** [reply]
Okay :)

**Comment by carlocab** [comment, marked as answer]
I think your problem should be fixed (for now) by

    brew reinstall -s gcc

It will take some time to complete, so it's probably best to run it overnight or something. I was able to reproduce the problem you had with `llvm`, and I found that rebuilding it from source fixed that.

**Reply by Brijesh1in** [reply]
Thanks a lot @carlocab . it worked ❤️

**Reply by carlocab** [reply]
Great! Thanks for the update. I was starting to wonder what had happened. Sorry it took so long!

**Reply by paulmccumber** [reply]
This worked for me as well.  I had the same issues with a fresh install of Homebrew and gcc.

**Comment by carlocab** [comment]
As an alternative to rebuilding `gcc` from source, you can also try

    export SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk

You may wish to add this to your `.zshrc` while bottles are being built on Big Sur 11.0.1.
